lib/util/native/get_destination_directory.dart

```diff
@@ -14,6 +16,12 @@ Future<String> getDefaultDestinationDirectory() async {
     case TargetPlatform.macOS:
     case TargetPlatform.windows:
     case TargetPlatform.fuchsia:
-      return (await path.getDownloadsDirectory())!.path.replaceAll('\\', '/');
+      var downloadDir = await path.getDownloadsDirectory();
+      if (defaultTargetPlatform == TargetPlatform.windows) {
```

Comment:
Thank you for this fix. I think there is a missing null check if `path.getDownloadsDirectory()` returns not null.

Also please check if the new path `~/Downloads` exists with `Directory(path).existsSync()` and fallback to `~`.

Reply:
Added them. The nullcheck got voided while the platform check happened